scripts/gen-doc.sh

```diff
@@ -28,39 +28,42 @@ DOCS_REPO="$(dirname "$SOF_REPO")/sof-docs"
 PUBLISH_REPO="$(dirname "$SOF_REPO")/thesofproject.github.io"
 
 # parse arguments
-DO_BUILD=no
-DO_CLEAN=no
-DO_PUBLISH=no
-DO_CLONE=no
-
-for args in $@
-do
-	if [[ "$args" == "-b" ]]
-	then
-		DO_BUILD=yes
-	elif [[ "$args" == "-c" ]]
-	then
-		DO_CLEAN=yes
-	elif [[ "$args" == "-p" ]]
-	then
-		DO_PUBLISH=yes
-	elif [[ "$args" == "-r" ]]
-	then
-		DO_CLONE=yes
-	fi
+options=$(getopt -n $0 -o bcpr -- "$@")
+
+eval set -- "$options"
+while true; do
+        case "$1" in
+        -b)
+                ACTION=build
```

Comment:
Might be a good idea to set some defaults for ACTION and FETCH_MISSING (for ACTION you can say "invalid" and check for it, for FETCH_MISSING set it as false)

Reply:
Done